assets/src/edit-story/output/page.js

```diff
@@ -73,13 +74,24 @@ function OutputPage({ page, autoAdvance, defaultPageDuration }) {
         <amp-story-grid-layer template="vertical" aspect-ratio={ASPECT_RATIO}>
           <div className="page-fullbleed-area">
             <div className="page-safe-area">
-              {regularElements.map((element) => (
-                <OutputElement key={'el-' + element.id} element={element} />
-              ))}
+              {regularElements.map((element) =>
+                /* Check for invalid elements, @todo this should come from the pre-publish checklist in the future. */
+                page.pageAttachment?.url?.length > 0 &&
+                isLinkBelowLimit(element) ? null : (
+                  <OutputElement key={'el-' + element.id} element={element} />
+                )
```

Comment:
Please filter the array beforehand as e.g.:

```js
const validElements = regularElements.filter((element) =>
  !hasPageAttachment || !isLinkBelowLimit(element)
);
...
validElements.map((element) => <OutputElement key={element.id} element={element} />;
```

Also note that you don't need the prefix for the `key` - keys are per-array only, not global for all children of any given node let alone any full component tree.

Reply:
Note: the `key` was there already before, tried to remove it, however, linter seems to complain, added it back for now.

Reply:
Oh, the key is necessary for sure, it just doesn't have to be prefixed as it was. `key={element.id}` is sufficient.